### PR TITLE
Spin button: ensure default fonts are set

### LINKED ISSUE #15362: SpinButton is not inheriting Fabric default fontFamily
### Environment Information

"office-ui-fabric-react": "^7.144.2",

### Please provide a reproduction of the bug in a codepen:

https://codepen.io/jacklovett/pen/bGebNPa

#### Actual behavior:

Since office-ui-fabric-react-7.140.0 when font-family: inherit was added to a number of input components here: https://github.com/microsoft/fluentui/pull/15172, SpinButton now displays as "Times" instead of the Fabric default font value. 

#### Expected behavior:

Expected behaviour is that SpinButton components would inherit the default Fabric font styles

### Priorities and help requested:

Are you willing to submit a PR to fix? No

Requested priority: (Low)

### PR DESCRIPTION
**Pull request by LeslieW**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #15362
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Currently, the SpinButton font family is not inherited from Fabric and defaults to 'Times'.

This change adds `fonts.medium` to the SpinButton component styles so that font family results in the Fabric default. This change mimics the use of font styling across other button types. 

Before:
spin_button_before

After:
spin_button_after

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by ecraig12345** [comment]
@xugao should probably review this since she made the change to add `font-family: inherit` in #15172.

File: apps/a11y-tests/src/tests/__snapshots__/ComponentExamples.test.tsx.snap [hunk 1]
```diff
@@ -1,6836 +0,0 @@
-// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (xugao) [inline apps/a11y-tests/src/tests/__snapshots__/ComponentExamples.test.tsx.snap:1]: pls revert this file

Comment (xugao) [reply apps/a11y-tests/src/tests/__snapshots__/ComponentExamples.test.tsx.snap:1]: other than that, the change looks good to me. `SpinButton` has always been relying on `Fabric` wrapper to drop the default fonts, which I think we should change to let `SpinButton` to define fonts on it own (your change)

Comment (LeslieW) [reply apps/a11y-tests/src/tests/__snapshots__/ComponentExamples.test.tsx.snap]: Thanks @xugao, done!

Comment (xugao) [reply apps/a11y-tests/src/tests/__snapshots__/ComponentExamples.test.tsx.snap]: looks like there are some snapshots you missed to update. To do that, go to `packages/react-examples` and run `yarn update-snapshots`

Comment (LeslieW) [reply apps/a11y-tests/src/tests/__snapshots__/ComponentExamples.test.tsx.snap]: Sounds good, thanks @xugao! Updated the snapshots.

**Review by xugao — APPROVED** [review]
LGTM! 

Could you please also make the same fix in `7.0` branch [here](https://github.com/microsoft/fluentui/blob/7.0/packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.styles.ts#L148). That way the fix will be published for  `office-ui-fabric-react@v7` / `@fluentui/react@v7` and you can pick up in your app right away.

The change you have now in master will be published under `@fluentui/react@v8-beta`.

**Review by khmakoto — APPROVED** [review]

**Comment by LeslieW** [comment]
Thanks @xugao and @khmakoto for your reviews! Sounds great, I created another [pull request](https://github.com/microsoft/fluentui/pull/15801) to add this change to the 7.0 branch.

**Pull request merged**